dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/vitals/VitalMonitor.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.vitals
+
+import kotlin.math.max
+import kotlin.math.min
+
+internal class VitalMonitor {
+
+    private var lastKnownSample: Double = Double.NaN
+
+    private val listeners: MutableMap<VitalListener, VitalInfo> = mutableMapOf()
```

Comment:
just a thought but instead of using the `synchronization` lock here maybe you could use a `ConcurrentHashMap<VitalListener,AtomicReference<VitalInfo>>` here.

Reply:
Using a ConcurrentHashMap might be an idea, but why the `AtomicReference` ?

Reply:
to handle the case where you set the value for a key from 2 different threads but I just had a look and the `ConcurrentHashMap` should handle that. I guess you do not need an `AtomicReference` there.

Reply:
`ConcurrentHashMap` doesn't lock at the retrieve process, so 2 threads may end up updating the same value, which is wrong behavior. But if we expect `onNewSample` to be called from the same thread always, this is fine.